gcal.ts

```diff
@@ -0,0 +1,68 @@
+const fetch = require('node-fetch');
+const status = require('http-status-codes');
+const querystring = require('querystring');
+import {unauthorizedError, userError} from './errors';
+
+export async function getEvents(calendarID: string, auth: string): Promise<Array<any>> {
+    const events = [];
+    let nextPageToken: string;
+
+    // get events from up to one year ago
+    const now = new Date();
+    const timeMin = `${now.getFullYear()-1}-${now.getMonth()+1}-${now.getDate()}T00:00:00Z`;
+
+    do {
+        const response = await gcal(
+            `/calendars/${encodeURIComponent(calendarID)}/events`,
+            auth,
+            {
+                qs: {
+                    maxResults: String(2500),
+                    pageToken: nextPageToken,
+                    timeMin,
+                }
+            });
+        response.items.forEach(item => {
+            item.calendarID = calendarID;
+        });
+        events.push(...response.items);
+        nextPageToken = response.nextPageToken;
+    } while (nextPageToken);
+
+    return events;
+}
+
+type GcalOptions = {
+    body?: string,
+    method?: string,
+    qs?: {[key: string]: string},
+}
+
+export async function gcal(path: string, auth: string, opts?: GcalOptions) {
+    if (!opts) opts = {};
+    const req = new fetch.Request(
```

Comment:
[cbb] `fetch(url, opts)` allows passing all that data without the extra `new fetch.Request`